2-mandatory/2-financial-times.js

```diff
@@ -23,16 +38,37 @@ function titleWithFewestWords(allArticleTitles) {
     (Hint: remember that you can also loop through the characters of a string if you need to)
 */
 function headlinesWithNumbers(allArticleTitles) {
-    // TODO
+    let numberArray =[]
+    for (item of allArticleTitles){
+        for (element of item){
+            element = parseInt(element)
+            if (item.includes(element)){
+            numberArray.push(item)
+            break
+            } 
+        }
+    }  
+ return numberArray;  
 }
 
 /*
     The Financial Times wants to understand what the average number of characters in an article title is.
     Implement the function below to return this number - rounded to the nearest integer.
 */
 function averageNumberOfCharacters(allArticleTitles) {
-    // TODO
-}
+    sumOfStrings=[]
+    for (item of allArticleTitles){
+        let itemLength = item.length
+       sumOfStrings.push(itemLength)
+       } 
+    const initialValue = 0;
+   const sumWithInitial = sumOfStrings.reduce(
```

Comment:
how would you do this without the reduce()?

Reply:
@llamojha I think I can make another while loop for sumOfStrings, which will sum all the elements in array. Do I need to change it?

Reply:
It's ok like this. Reduce() will work perfectly in this case. But because the exercise is about loops, it is also good to show how it will look if you use a loop instead. We can go through this in our next buddy session.